src/hotspot/share/oops/fieldInfo.inline.hpp

```diff
@@ -151,23 +152,12 @@ inline FieldInfoReader& FieldInfoReader::set_position_and_next_index(int positio
 }
 
 inline void FieldStatus::atomic_set_bits(u1& flags, u1 mask) {
-  // Atomically update the flags with the bits given
-  u1 old_flags, new_flags, witness;
-  do {
-    old_flags = flags;
-    new_flags = old_flags | mask;
-    witness = Atomic::cmpxchg(&flags, old_flags, new_flags);
-  } while (witness != old_flags);
+  Atomic::fetch_then_or(&flags, mask);
 }
 
 inline void FieldStatus::atomic_clear_bits(u1& flags, u1 mask) {
-  // Atomically update the flags with the bits given
-  u1 old_flags, new_flags, witness;
-  do {
-    old_flags = flags;
-    new_flags = old_flags & ~mask;
-    witness = Atomic::cmpxchg(&flags, old_flags, new_flags);
-  } while (witness != old_flags);
+  u1 val = (~mask);
+  Atomic::fetch_then_and(&flags, val);
```

Comment:
u1 is not a supported type for Atomic bitops.  This only happens to work right now because all
platforms are currently using a cmpxchg-based implementation and aren't enforcing the documented
limitation of only providing support for size of an int or size of a pointer (if different).

Reply:
But I need u1! I thought that was the point of having the templates?  Do I change this back to my own CAS loop?